xir/program.py

```diff
@@ -137,27 +137,44 @@ def use_floats(self) -> bool:
         return self._use_floats
 
 
+class ObservableFactor:
+    def __init__(self, name: str, params: [], wires: [int]):
+        self.name = name
+        self.params = params
+        self.wires = wires
+
+    def __str__(self):
+        if len(self.params) == 0:
+            return f"{self.name}{self.wires}"
+        else:
+            return f"{self.name}({', '.join([str(v) for v in self.params])}){self.wires}"
```

Comment:
I like one-liners as much as the next person but I think it makes the code less readable in this case:
```suggestion
            params = "(" + ", ".join(map(str, self.params))} + ")"
            return f"{self.name}{params}{self.wires}"
```